src/remove-cues-from-track.js

```diff
@@ -27,8 +27,8 @@ const removeCuesFromTrack = function(start, end, track) {
   while (i--) {
     cue = track.cues[i];
 
-    // Remove any overlapping cue
-    if (cue.startTime <= end && cue.endTime >= start) {
+    // Remove any cue wholly within the removal region
+    if (cue.startTime >= start && cue.endTime <= end) {
```

Comment:
Yes. If you're using the cue metadata to determine buffer size, then you must not remove cues whose corresponding segments are only being partially removed, otherwise you'd end up with an estimate that undershoots the actual value.

Reply:
I see! I believe this method is also used for textTracks - in that scenario would leaving the overlapping cues cause strange issues?

Reply:
IIRC, there are two ways this function is called:
1) bounded `(0, currentTime() - backBufferLength)`, where it will remove cues a certain amount of time behind the playhead
2) unbounded `(0, duration())`, when it resets the stream.

In this way, only removing cues that wholly overlap with the bounded removes is probably better, in that you won't accidentally remove a cue that starts before the buffer does but ends after. i.e. if you have a cue (13, 15) and a remove is called for (0, 14), you should be displaying that cue from 14s->15s if you rewind the playhead to 14s.

Reply:
Actually, I thought of a potential problem. Cues in VTT playlists don't necessarily have to end at the duration of the buffer -- they could end after. So a call to remove `(0, duration())` may not catch a cue that hangs over the end of the video.

Reply:
Did you mean the duration of the buffered region or the whole video's duration? I just double-checked and it looks like WebVTT cues don't necessarily need to end at the video duration so you're right that we could potentially miss a cue hanging over the end of the video and never clear it. Perhaps it would be worth having another utility function to clear all the cues from beginning to end specifically for visible textTracks?

Reply:
> we could potentially miss a cue hanging over the end of the video and never clear it.

This is what I was talking about.

> Perhaps it would be worth having another utility function to clear all the cues from beginning to end specifically for visible textTracks?

And just do like a `textTrack.cues.forEach((cue) => textTrack.remove(cue))` kind of thing you mean?

Reply:
yeah, essentially just `removeAllCuesFromTrack(track)` that does `textTrack.cues.forEach((cue) => textTrack.remove(cue))`

Reply:
I added a `removeAllCuesFromTrack`.

I guess one other option would be to simply pass `Infinity` other than `duration()` for the call to remove all text cues inside contrib-hls, since all cues should fall within [0, Infinity].

Reply:
That would also make sense!